invenio_search/__init__.py

```diff
@@ -22,7 +22,263 @@
+forgot to create the index (follow the steps from :ref:`creating_index` to
+see how to setup an index and add example data).
+
+
+Creating a search page
+~~~~~~~~~~~~~~~~~~~~~~
+
+Let's create a simple web page where you can send queries to the Elasticsearch
+and see the results.
+Create a new app.py file with a route.
+
+.. code-block:: python
+
+    # app.py
+    from elasticsearch_dsl.query import QueryString
+    from flask import Flask, jsonify, request
+    from invenio_search import InvenioSearch, RecordsSearch
+
+    app = Flask('myapp')
+
+    search = InvenioSearch(app)
+
+    # This line is needed to be able to call `flask index init`
+    search.register_mappings('demo', 'examples.data.elastic_v2')
+
+
+    @app.route('/', methods=['GET', 'POST'])
+    def index():
+        search = RecordsSearch()
+        if 'q' in request.values:
+            search = search.query(QueryString(query=request.values.get('q')))
+
+        return jsonify(search.execute().to_dict())
+
+Run example development server:
+
+.. code-block:: console
+
+   $ FLASK_DEBUG=1 FLASK_APP=app.py flask run -p 5000
+
+And now you can perform search queries in the browser:
+
+.. code-block:: console
+
+   $ open http://localhost:5000/?q=body:test
+
+Filtering
+~~~~~~~~~
+
+To filter out some documents, you can create your own search class. Let's try
+to remove all private documents from the search results (by ``private``
+documents, we understand all the documents that have ``public`` attribute set
+to 0).
+
+First, let's add one public and one private document to Elasticsearch:
+
+.. code-block:: console
+
+    $ echo '{"title": "Public", "body": "test 1", "public": 1}' | \
+        flask index put demo-default-v1.0.0 example
+    $ echo '{"title": "Private", "body": "test 2", "public": 0}' | \
+        flask index put demo-default-v1.0.0 example
+
+Now, create a new search class that will return all documents of type
+``example`` from the ``demo`` index and select only the public ones
+(documents where ``public`` is set to 1):
+
+.. code-block:: python
+
+    # app.py
+    from elasticsearch_dsl.query import Bool, Q, QueryString
+
+    class PublicSearch(RecordsSearch):
+        class Meta:
+            index = 'demo'
+            doc_types = ['example']
+            fields = ('*', )
+            facets = {}
+
+        def __init__(self, **kwargs):
+            super(PublicSearch, self).__init__(**kwargs)
+            self.query = self.query._proxied & Q(
+                Bool(filter=[Q('term', public=1)])
+            )
+
+Update the ``index`` function and replace the search class with our new
+``PublicSearch`` class:
+
+.. code-block:: python
+
+    # app.py
+    @app.route('/', methods=['GET', 'POST'])
+    def index():
+        search = PublicSearch()
+        ...
+
+Now, open the browser and search for documents with ``test`` in the body.
+
+.. code-block:: console
+
+   $ open http://localhost:5000/?q=body:test
```

Comment:
doesn't this alias work everywhere? Shall we add something more generic like ``curl``? :)

Reply:
That's a good point and I think the problem is bigger than this one example:
```console
$ ack "open http" .
new-cookiecutter/invenio-search/examples/app.py
50:    $ open http://127.0.0.1:5000/

invenio-admin/examples/app.py
50:    $ open http://127.0.0.1:5000/

elasticsearch/elasticsearch-head/README.textile
21:* @open http://localhost:9200/_plugin/head/@

invenio-oauth2server/examples/consumer.py
44:   $ open http://127.0.0.1:5000/account/settings/applications
79:    $ open http://127.0.0.1:5000/logout

invenio-oauth2server/examples/app.py
44:   $ open http://127.0.0.1:5000/account/settings/applications

invenio-search/examples/app.py
51:   $ open http://localhost:5000/?q=body:test

invenio-search/invenio_search/__init__.py
202:   $ open http://localhost:5000/?q=body:test
258:   $ open http://localhost:5000/?q=body:test

invenio-accounts/examples/app.py
55:    $ open http://127.0.0.1:5000/

invenio-access/examples/app.py
54:    $ open http://127.0.0.1:5000/

invenio-userprofiles/examples/app.py
54:    $ open http://127.0.0.1:5000/
63:    $ open http://127.0.0.1:5000/admin

invenio-pidstore/examples/app.py
51:    $ open http://0.0.0.0:5000/admin/recordmetadata/

cookiecutter-invenio-module/{{ cookiecutter.project_shortname }}/examples/app.py
27:    $ open http://127.0.0.1:5000/

```
I think we should decide what to do with this and change it in the `cookiecutter` (and other modules). I'm fine with curl.

Reply:
I don't have much trouble to understand `open`. Besides it's pro-macos :)

Reply:
Lol, that's my wifes account (im jbenito3 from my stupid phone)

Reply:
Hahahahahahahahahahahaha :laughing: 
That really made my evening :D
I was wondering why some random people are commenting on documentation of `invenio-search` module :D

Reply:
I have replaced `open` with `curl`

Reply:
so I see that you didn't care about my wife's opinion :)

Reply:
The fact that we use `curl` more often than `open` was the deciding factor :) Also @drjova doesn't have the same aliases as you, as he can't use `open` on his Macbook. Probably it's a part of Apple's strategy to remove useful things from Macbooks with every new generation ;)